test/SpeedJumpIrmTest.sol

```diff
@@ -121,6 +121,90 @@ contract AdaptativeCurveIrmTest is Test {
         assertApproxEqRel(irm.borrowRateView(marketParams, market), uint256(0.00057 ether) / 365 days, 0.1 ether);
     }
 
+    function testRateAfter60DaysUtilizationAboveTargetNoPing() public {
+        Market memory market;
+        market.totalSupplyAssets = 1 ether;
+        market.totalBorrowAssets = uint128(uint256(TARGET_UTILIZATION));
+        assertEq(irm.borrowRate(marketParams, market), uint256(INITIAL_RATE_AT_TARGET));
+        assertEq(irm.rateAtTarget(marketParams.id()), INITIAL_RATE_AT_TARGET);
+
+        market.lastUpdate = uint128(block.timestamp);
+        vm.warp(block.timestamp + 60 days);
+
+        market.totalBorrowAssets = uint128(uint256(TARGET_UTILIZATION + 1 ether) / 2);
+        irm.borrowRate(marketParams, market);
+
+        assertApproxEqRel(irm.rateAtTarget(marketParams.id()), int256(0.6092 ether) / 365 days, 0.0001 ether);
+    }
+
+    function testRateAfter60DaysUtilizationAboveTargetPingEveryHour() public {
+        Market memory market;
+        market.totalSupplyAssets = 1 ether;
+        market.totalBorrowAssets = uint128(uint256(TARGET_UTILIZATION));
+        assertEq(irm.borrowRate(marketParams, market), uint256(INITIAL_RATE_AT_TARGET));
+        assertEq(irm.rateAtTarget(marketParams.id()), INITIAL_RATE_AT_TARGET);
+
+        market.totalBorrowAssets = uint128(uint256(TARGET_UTILIZATION + 1 ether) / 2); // Error = 50%
+
+        for (uint256 i; i < 60 days / 1 hours; ++i) {
+            market.lastUpdate = uint128(block.timestamp);
+            vm.warp(block.timestamp + 1 hours);
+
+            uint256 avgBorrowRate = irm.borrowRate(marketParams, market);
+            uint256 interest = market.totalBorrowAssets.wMulDown(avgBorrowRate.wTaylorCompounded(1 hours));
+            market.totalSupplyAssets += uint128(interest);
+            market.totalBorrowAssets += uint128(interest);
+        }
+
+        assertApproxEqRel(market.totalBorrowAssets.wDivDown(market.totalSupplyAssets), 0.95 ether, 0.005 ether);
+        assertApproxEqRel(irm.rateAtTarget(marketParams.id()), int256(0.6092 ether) / 365 days, 0.06 ether);
```

Comment:
We could add a check on the total borrow assets, that the error is not bigger than x%

Reply:
Marked as resolved because you don't think it's necessary, or it was not pushed ?